de.fhdo.lemma.servicedsl.extractor/src/de/fhdo/lemma/servicedsl/extractor/ServiceDslExtractor.xtend

```diff
@@ -0,0 +1,202 @@
+package de.fhdo.lemma.servicedsl.extractor
+
+import de.fhdo.lemma.data.ComplexType
+import de.fhdo.lemma.data.PrimitiveType
+import de.fhdo.lemma.service.Endpoint
+import de.fhdo.lemma.service.Import
+import de.fhdo.lemma.service.ImportType
+import de.fhdo.lemma.service.ImportedType
+import de.fhdo.lemma.service.Interface
+import de.fhdo.lemma.service.Microservice
+import de.fhdo.lemma.service.MicroserviceType
+import de.fhdo.lemma.service.Operation
+import de.fhdo.lemma.service.Parameter
+import de.fhdo.lemma.service.ServiceModel
+import de.fhdo.lemma.service.Visibility
+import de.fhdo.lemma.technology.CommunicationType
+import de.fhdo.lemma.technology.ExchangePattern
+import de.fhdo.lemma.technology.Protocol
+import de.fhdo.lemma.service.ImportedServiceAspect
+
+class ServiceDslExtractor {
+    static val IMPORTED_TECH_ALIASES = <String>newArrayList()
+    static val ID_PATTERN = "(\\^?)([a-zA-Z_])\\w*"
+    static val QUALIFIED_NAME_PATTERN = '''«ID_PATTERN»(\.«ID_PATTERN»)*'''
+    static val QUALIFIED_NAME_WITH_AT_LEAST_ONE_LEVEL_PATTERN = '''«ID_PATTERN»\.''' +
+        QUALIFIED_NAME_PATTERN
+
+    /**
+     * Extract ServiceModel
+     */
+    def String extractToString(ServiceModel serviceModel) {
+        val imports = <String>newArrayList
+        serviceModel.imports.forEach[imports.add(it.generate)]
+        val importStatements = if (!imports.empty)
+                String.join("\n", imports) + "\n\n"
+            else
+                ""
+
+        val microservices = String.join("\n\n", serviceModel.microservices.map[generate])
+
+        return '''«importStatements»«microservices»'''
+    }
+
+    private def generate(Import ^import) {
+        val importTypeKeyword = switch(^import.importType) {
+            case DATATYPES: "datatypes"
+            case MICROSERVICES: "microservices"
+            case TECHNOLOGY: "technology"
+            default: null
+        }
+        //if it is a technology add it to the names list for later use in the generation
+        if(importTypeKeyword === "technology") IMPORTED_TECH_ALIASES.add(^import.name)
```

Comment:
Don't use reference equality to compare strings (it works here because `importTypeKeyword` received the literal `"technology"` with which it is compared here but usually this is not the case; see also the transpiled Java code for reference https://github.com/SeelabFhdo/lemma/blob/120ecba6bc044cf89519d5504860126419699131/de.fhdo.lemma.servicedsl.extractor/xtend-gen/de/fhdo/lemma/servicedsl/extractor/ServiceDslExtractor.java#L117). Instead, use Xtend's `==`-operator, which transpiles to `equals()`, which will succeed even if you do not compare the same literals but values within different `String` instances.

Also: Put the body of the `if`-statement in the next line.